### PR TITLE
(PUP-4033) Test for correct puppet configuration

### PR DESCRIPTION
**Pull request by MikaelSmith**
Tests that the results of `puppet agent --genconfig` and `puppet agent
--configprint` match puppet-specifications, and that files that are
expected to exist from the installer and agent runs are in the locations
specified in said specification.

### CODE REVIEW

File: acceptance/tests/ensure_puppet-agent_paths.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      end
+    when :file
+      if !file_exists?(agent, path)
+        fail_test("Failed to find expected file '#{path}' on agent '#{agent}'")
+      end
+    end
+  end
+end
+
+step 'test puppet binaries exist'
```
Comment (MikaelSmith) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: @er0ck Thoughts on testing this? Binaries for AIO vs source builds will be in different places.

**Comment by MikaelSmith** [comment]
Passes ci:test:git and ci:test:aio on Linux, ci:test:git on Windows. Still pending ci:test:aio run with Windows.

File: acceptance/tests/ensure_puppet-agent_paths.rb (hunk 1 above)
Comment (er0ck) [inline acceptance/tests/ensure_puppet-agent_paths.rb]: yeah.  i was concerned about that.  is there a way we can tell if we're in an AIO or source produced environment?
oh: duh  option[:type]  should tell us if we're aio, git, package, etc

**Comment by MikaelSmith** [comment]
Passes Windows AIO run with the test for a file existing at :rest_authconfig removed. That change makes sense, since it's not setup by installation.

**Comment by er0ck** [comment]
(y) if this is ready to go

**Comment by MikaelSmith** [comment]
Should be ready.

**Pull request merged**